sourcegraph/sourcegraph
# Cody JetBrains: add completions

Previously, the Cody JetBrains plugin only supported Chat. This PR adds infrastructure to additionally provide inline completions using JetBrains `Inlay` API. The current implementation is minimal and provides a fairly bad user experience. The primary goal of this PR is to add the infrastructure so that we can tweak and improve the UX with smaller PRs.

## Test plan

- Green CI
- Manual testing:
  - Validate that you are logged into sourcegraph.com with `src login`. This is required since completions use the environment variable `SRC_ACCESS_TOKEN` for now.
  - Run `./gradlew :runIde`
  - Open a random Java codebase, for example https://github.com/spotify/dns-java (don't use the scip-java repo because it requires the Scala plugin to load correctly.
  - Try editing Java code and see the completions appear

![CleanShot 2023-05-29 at 21 33 11](https://github.com/sourcegraph/sourcegraph/assets/1408093/12288d07-053e-4426-9129-c800cd7c9bc3)
